src/Raven.Client/Util/CertificateLoaderUtil.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Diagnostics;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+
+namespace Raven.Client;
+
+internal static class CertificateLoaderUtil
+{
+    public static X509KeyStorageFlags FlagsForExport => X509KeyStorageFlags.Exportable;
+
+    public static X509KeyStorageFlags FlagsForPersist => X509KeyStorageFlags.Exportable | X509KeyStorageFlags.PersistKeySet;
+
+    public static void Import(X509Certificate2Collection collection, byte[] rawData, string password = null, X509KeyStorageFlags? flags = null)
+    {
+        Import(f => collection.Import(rawData, password, f), flags);
+    }
+    internal static void Import(X509Certificate2Collection collection, string fileName, string password = null, X509KeyStorageFlags? flags = null)
+    {
+        Import(f => collection.Import(fileName, password, f), flags);
+    }
+    
+    private static void Import(Action<X509KeyStorageFlags> importer, X509KeyStorageFlags? flag)
+    {
+        DebugAssertDoesntContainKeySet(flag);
+        var f = AddUserKeySet(flag);
+        try
+        {
+            importer(f);
+        }
+        catch
+        {
+            importer(SwitchUserKeySetWithMachineKeySet(f));
+        }
+    }
+    
+    public static X509Certificate2 CreateCertificate(byte[] rawData, string password = null, X509KeyStorageFlags? flags = null)
+    {
+        return CreateCertificate(f => new X509Certificate2(rawData, password, f), flags);
+    }
+    internal static X509Certificate2 CreateCertificate(string fileName, string password = null, X509KeyStorageFlags? flags = null)
+    {
+        return CreateCertificate(f => new X509Certificate2(fileName, password, f), flags);
+    }
+
+    private static X509Certificate2 CreateCertificate(Func<X509KeyStorageFlags, X509Certificate2> creator, X509KeyStorageFlags? flag)
+    {
+        DebugAssertDoesntContainKeySet(flag);
+        var f = AddUserKeySet(flag);
+        try
+        {
+            return creator(f);
+        }
+        catch
+        {
```

Comment:
can we log here? (but once)

Reply:
Yes, sure.
Maybe worth to log on the Operation level for the first time and on the Information level for the rest.

Reply:
done